File: Oidc.AuthorizationServer/Models/ClaimEntity.cs
using FluentNHibernate.Data;
using lvl.Ontology;
using lvl.Ontology.Authorization;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Security.Claims;

namespace lvl.Oidc.AuthorizationServer.Models
{
    /// <summary>
    ///     Allows a claim to be stored in the database.
    /// </summary>
    [Table(nameof(Claim), Schema = "oidc")]
    [HiddenFromApi]
    public class ClaimEntity : Entity, IAggregateRoot
    {
        /// <summary>
        ///     Gets the claim type of the claim.
        /// </summary>
        [Required]
        public string Type { get; set; }

        /// <summary>
        ///     Gets the issuer of the claim.
        /// </summary>
        public string Issuer { get; set; }

        /// <summary>
        ///     Gets the value type of the claim.
        /// </summary>
        public string ValueType { get; set; }

        /// <summary>
        ///     Gets the value of the claim.
        /// </summary>
        [Required]
        public string Value { get; set; }

        public Claim ToSecurityClaim()
        {
            return new Claim(Type, Value, ValueType, Issuer);
        }
    }
}

File: oidc/src/models/claim-entity.ts
import { Entity, IAggregateRoot } from '@lvl/front-end';

export class ClaimEntity extends Entity implements IAggregateRoot {
    public type: string;
    public issuer: string;
    public valueType: string;
    public value: string;

    constructor(options?: IClaimEntityOptions) {
        options = options || {};
        super(options);
        this.type = options!.type;
        this.issuer = options!.issuer;
        this.valueType = options!.valueType;
        this.value = options!.value;
    }
}

interface IClaimEntityOptions {
    type?: string;
    issuer?: string;
    valueType?: string;
    value?: string;
    id?: number;
}
